plugins/basic/nexus-repository-nuget/src/test/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetEntryTest.java

```diff
@@ -50,19 +52,24 @@ public void packageEntrySmokeTest() throws Exception {
     final String packageId = "screwdriver";
     final String version = "0.1.1";
 
-    final StorageTx tx = mock(StorageTx.class);
-    doReturn(tx).when(underTest).openStorageTx();
-
     // Wire the mocks together: component has asset, asset has format attributes
-    doReturn(component).when(underTest).findComponent(tx, packageId, version);
-    doReturn(asset).when(underTest).findAsset(tx, component);
+    doReturn(component).when(underTest).findComponent(any(StorageTx.class), eq(packageId), eq(version));
+    doReturn(asset).when(underTest).findAsset(any(StorageTx.class), eq(component));
     doReturn(mock(NestedAttributesMap.class)).when(asset).formatAttributes();
 
     final HashMap<String, ?> data = Maps.newHashMap();
     doReturn(data).when(underTest).toData(any(NestedAttributesMap.class),
         anyMapOf(String.class, String.class));
 
-    underTest.entry("base", packageId, version);
+    final StorageTx tx = mock(StorageTx.class);
+
+    UnitOfWork.beginBatch(tx);
```

Comment:
Yes, that's right - unless we use powermock to intercept the static methods. I'm ok with renaming this method if it would help... I used "beginBatch" to match the other case where a single TX is maintained across the work, but other suggestions are welcome.